src/Build.UnitTests/Evaluation/ProjectCollection_Tests.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using Microsoft.Build.Evaluation;
+using Shouldly;
+using Xunit;
+
+namespace Microsoft.Build.UnitTests.Evaluation
+{
+    public class ProjectCollection_Tests
+    {
+        [Fact]
+        public void ProjectRootElementCache_IsDeterminedByEnvironmentVariable()
```

Comment:
Maybe add another one liner test which asserts that the beefy implementation is used when the env var is not set :)

Reply:
Ah, nevermind, the existing test does both :P